### PR TITLE
Forward error to done()

### PR DESCRIPTION
**Pull request by johanneswuerbach**
https://github.com/vitaly-t/pg-promise/pull/573 actually didn't properly fix it. This is now the real fix and a test-case that it works.

### CODE REVIEW

File: test/dbSpec.js [hunk 1]
```diff
@@ -329,6 +329,64 @@ describe('Connection', () => {
...
+        let error;
+
+        beforeEach(done => {
+            dbSingleCN.connect()
+                .then(obj => {
+
+                    const q = obj.any('SELECT pg_sleep(1);');
+
+                    // Terminate all connections after a short delay
+                    promise.delay(100).then(() => {
```
Comment (johanneswuerbach) [inline test/dbSpec.js]: If the `delay(100)` is removed this still fails, but that seems to be a different bug :-(

**Comment by vitaly-t** [comment]
@johanneswuerbach As you can see, your changes result un errors. Do you want to re-do it?

**Comment by johanneswuerbach** [comment]
@vitaly-t updated.

**Comment by vitaly-t** [comment]
@johanneswuerbach I had to comment out most of the test, as it is flimsy, works only sometimes.

If you get a chance to rewrite the test, please do another PR.

**Comment by johanneswuerbach** [comment]
@vitaly-t yes, there seems to be actually more then one issue with error handling, but so far I wasn't able to pin it down.

It seems to be that sometimes `ctx` is disconnected multiple times and sometimes the `onError` handler is called before going into query causing weird states.

**Pull request merged**